torch/ao/quantization/quantizer/utils.py

```diff
@@ -47,3 +60,55 @@ def _node_only_used_for_sym_size(node: Node, partition_nodes: List[Node]):
         ((user not in partition_nodes) or _is_sym_size_node(user))
         for user in node.users
     )
+
+
+def _get_conv_unary_pattern(
+    conv_fn: Callable,
+    has_bn: bool = False,  # Usually need for QAT pattern
+    unary_fn: Optional[Callable[[Any], Any]] = None,
+):
+    def _conv_unary(
+        x,
+        conv_weight,
+        conv_bias,
+        bn_weight=None,
+        bn_bias=None,
+        bn_rm=None,
+        bn_rv=None,
+    ):
+        conv = conv_fn(x, conv_weight, conv_bias)
+        if has_bn:
+            bn = F.batch_norm(conv, bn_rm, bn_rv, bn_weight, bn_bias, training=True)
+        else:
+            bn = conv
+        if unary_fn is not None:
+            output = unary_fn(bn)
+        else:
+            output = bn
+        return output, {
+            "input": x,
+            "conv": conv,
+            "conv_weight": conv_weight,
+            "conv_bias": conv_bias,
+            "output": output,
+        }
+
+    return _conv_unary
+
+
+def _generate_pattern_matcher_helper(pattern, example_inputs):
+    pattern = get_aten_graph_module(pattern, example_inputs)
+    pattern.graph.eliminate_dead_code()
+    pattern.recompile()
+    return SubgraphMatcherWithNameNodeMap(pattern, ignore_literals=True)
+
+
+@functools.lru_cache(None)
+def _generate_conv2d_pattern_matcher():
+    # Ensure it's only be invoked once, due to the cache size limitation in
+    # https://github.com/pytorch/pytorch/blob/
+    # 4c6e842496da636123f83ef868ca1974631f1f1e/torch/_dynamo/config.py#L35-L39
+    return _generate_pattern_matcher_helper(
+        _get_conv_unary_pattern(torch.nn.functional.conv2d),
```

Comment:
How is `torch.nn.Conv2d` handled?

Reply:
I think `torch.nn.Conv2d` and `torch.nn.functional.conv2d` will generate same node `torch.ops.aten.conv2d.default` after export. So, we should be fine to only use `torch.nn.functional.conv2d` here.